src/components/CreateTodo.vue

```diff
@@ -73,23 +81,23 @@ export default {
   },
   mounted() {
     if (this.inDetailMode) {
-      const { todo, status } = this.todo
-
-      if (status === "ok") {
-        this.title = todo.title
-        this.description = todo.desc
-      }
+      this.title = this.todoItem.title
+      this.description = this.todoItem.desc
     }
   },
   methods: {
-    onAdd() {
+    async onAdd() {
       if (!this.isValidate()) return
 
-      this.$store.dispatch("addTodoItem", {
+      this.showSpinner = true
+
+      await this.$store.dispatch("addTodoItem", {
         title: this.title,
         desc: this.description,
       })
 
+      this.showSpinner = false
+
       this.$emit("add")
       this.onCancel()
     },
```

Comment:
Use try, catch and finally block. In the finally block do whatever need to do after the API call (i.e `this.showSpinner = false`). If anything come from catch, show a error message.